src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedRecordPropertySymbol.cs

```diff
@@ -0,0 +1,210 @@
+                this,
+                GeneratedNames.MakeBackingFieldName(name),
+                isReadOnly: true,
+                isStatic: false,
+                hasInitializer: backingParameter.HasExplicitDefaultValue);
+            GetMethod = new GetAccessorSymbol(this, name);
+        }
+
+        internal override bool IsAutoProperty => true;
+
+        public override RefKind RefKind => RefKind.None;
+
+        public override TypeWithAnnotations TypeWithAnnotations => _backingParameter.TypeWithAnnotations;
+
+        public override ImmutableArray<CustomModifier> RefCustomModifiers => ImmutableArray<CustomModifier>.Empty;
+
+        public override ImmutableArray<ParameterSymbol> Parameters => ImmutableArray<ParameterSymbol>.Empty;
+
+        public override bool IsIndexer => false;
+
+        public override MethodSymbol GetMethod { get; }
+
+        public override MethodSymbol? SetMethod => null;
+
+        public override ImmutableArray<PropertySymbol> ExplicitInterfaceImplementations => ImmutableArray<PropertySymbol>.Empty;
+
+        public override Symbol ContainingSymbol => _containingType;
+
+        public override ImmutableArray<Location> Locations => _backingParameter.Locations;
+
+        public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences => _backingParameter.DeclaringSyntaxReferences;
+
+        public override Accessibility DeclaredAccessibility => Accessibility.Public;
+
+        public override bool IsStatic => false;
+
+        public override bool IsVirtual => false;
+
+        public override bool IsOverride => false;
+
+        public override bool IsAbstract => false;
+
+        public override bool IsSealed => false;
+
+        public override bool IsExtern => false;
+
+        internal override bool HasSpecialName => false;
+
+        internal override CallingConvention CallingConvention => CallingConvention.HasThis;
+
+        internal override bool MustCallMethodsDirectly => false;
+
+        internal override ObsoleteAttributeData? ObsoleteAttributeData => null;
+
+        public override string Name => _backingParameter.Name;
+
+        protected override IAttributeTargetSymbol AttributesOwner => this;
+
+        protected override AttributeLocation AllowedAttributeLocations => AttributeLocation.None;
+
+        protected override AttributeLocation DefaultAttributeLocation => AttributeLocation.None;
+
+        public override ImmutableArray<CSharpAttributeData> GetAttributes() => ImmutableArray<CSharpAttributeData>.Empty;
+
+        internal override bool HasPointerType => Type.IsPointerType();
+
+        public override SyntaxList<AttributeListSyntax> AttributeDeclarationSyntaxList => new SyntaxList<AttributeListSyntax>();
+
+        private sealed class GetAccessorSymbol : SynthesizedInstanceMethodSymbol
+        {
+            private readonly SynthesizedRecordPropertySymbol _property;
+
+            public override string Name { get; }
+
```

Comment:
Consider moving below constructor.

Reply:
Do you mind if I do the opposite? Now that I'm looking at it, I find it much more useful to group all the auto-properties that are getting initialized in the constructor at the top.

I'm not sure why we have a convention that differs between fields and auto-properties, because they're basically the same thing.